src/main/java/net/fabricmc/stitch/commands/CommandGenerateIntermediary.java

```diff
@@ -47,8 +48,44 @@ public void run(String[] args) throws Exception {
             e.printStackTrace();
         }
 
-        System.err.println("Generating new mappings...");
         GenState state = new GenState();
+        boolean clearedPatterns = false;
+
+        for (int i = 2; i < args.length; i++) {
+            if (args[i].startsWith("--")) {
+                switch (args[i].substring(2).toLowerCase(Locale.ROOT)) {
```

Comment:
Could you please rework this so that the logic for the arguments is not duplicated?